connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -181,7 +181,7 @@ public SchemaBuilder doc(String doc) {
 
     @Override
     public Map<String, String> parameters() {
-        return Collections.unmodifiableMap(parameters);
+        return parameters == null ? null: Collections.unmodifiableMap(parameters);
```

Comment:
Do you want to return `null` or an empty collection when `parameters` is `null`?